search/src/main/scala/weco/api/search/services/WorksRequestBuilder.scala

```diff
@@ -179,8 +179,28 @@ object WorksRequestBuilder
           field = "data.items.locations.locationType.id",
           values = itemLocationTypeIds
         )
-      case PartOfFilter(id) =>
-        termQuery(field = "state.relations.ancestors.id", value = id)
+      case PartOfFilter(search_term) =>
```

Comment:
I think this whole match block is ripe for refactoring, but I won't do it now as part of this change.

It basically amounts to a chain of `if object is of class X, do this else if object is of class Y do that`.

I think it would be better for each of these cases to be a method

Reply:
I would be slightly hesitant about that sort of change; we used to have a very DRY and modular class for building queries, and nobody - especially especially non-Scala devs - could reason about how it related to the JSON that elastic4s produces. 

The advantage of doing it like this is that it looks pretty much like the JSON that it produces, and it's easier for eg @harrisonpim to make changes. I think we're gravitating towards the view that the API shouldn't be written in Scala for exactly this sort of reason, and that some sort of language that is better at handling JavaScript Object Notation might be more appropriate and work better for the team...

Reply:
I would be slightly hesitant about that sort of change; we used to have a very DRY and modular class for building queries, and nobody - especially especially non-Scala devs - could reason about how it related to the JSON that elastic4s produces. 

The advantage of doing it like this is that it looks pretty much like the JSON that it produces, and it's easier for eg @harrisonpim to make changes. I think we're gravitating towards the view that the API shouldn't be written in Scala for exactly this sort of reason, and that some sort of language that is better at handling JavaScript Object Notation might be more appropriate and work better for the team...

Reply:
Fair enough.  I guessed that there would be a reason behind it.  I just find this pattern a bit odd

Shifting to another language would also help to remove the internal_model dependency.  If we can make it so it returns data (almost) verbatim from the database, then may not even need types.